hs_modflow_modelinstance/models.py

```diff
@@ -0,0 +1,799 @@
+        nam_file_count = 0
+        existing_files = []
+        reqd_files = []
+        for res_file in self.files.all():
+                ext = os.path.splitext(res_file.resource_file.name)[-1]
+                existing_files.append(res_file.resource_file.name.split("/")[-1])
+                if ext == '.nam':
+                    nam_file_count += 1
+                    name_file = res_file.resource_file.file
+                    for rows in name_file:
+                        rows = rows.strip()
+                        rows = rows.split(" ")
+                        r = rows[0].strip()
+                        if not r.startswith('#') and r != '' and r.lower() != 'list' \
+                                and r.lower() != 'data' and r.lower() != 'data(binary)':
+                            reqd_files.append(rows[-1].strip())
+        return nam_file_count, reqd_files, existing_files
+
+
+processor_for(MODFLOWModelInstanceResource)(resource_processor)
+
+
+# metadata container class
+class MODFLOWModelInstanceMetaData(ModelInstanceMetaData):
+    _study_area = GenericRelation(StudyArea)
+    _grid_dimensions = GenericRelation(GridDimensions)
+    _stress_period = GenericRelation(StressPeriod)
+    _ground_water_flow = GenericRelation(GroundWaterFlow)
+    _boundary_condition = GenericRelation(BoundaryCondition)
+    _model_calibration = GenericRelation(ModelCalibration)
+    _model_input = GenericRelation(ModelInput)
+    _general_elements = GenericRelation(GeneralElements)
+
+    @property
+    def study_area(self):
+        return self._study_area.all().first()
+
+    @property
+    def grid_dimensions(self):
+        return self._grid_dimensions.all().first()
+
+    @property
+    def stress_period(self):
+        return self._stress_period.all().first()
+
+    @property
+    def ground_water_flow(self):
+        return self._ground_water_flow.all().first()
+
+    @property
+    def boundary_condition(self):
+        return self._boundary_condition.all().first()
+
+    @property
+    def model_calibration(self):
+        return self._model_calibration.all().first()
+
+    @property
+    def model_inputs(self):
+        return self._model_input.all()
+
+    @property
+    def general_elements(self):
+        return self._general_elements.all().first()
+
+    @classmethod
+    def get_supported_element_names(cls):
+        # get the names of all core metadata elements
+        elements = super(MODFLOWModelInstanceMetaData, cls).get_supported_element_names()
+        # add the name of any additional element to the list
+        elements.append('StudyArea')
+        elements.append('GridDimensions')
+        elements.append('StressPeriod')
+        elements.append('GroundWaterFlow')
+        elements.append('BoundaryCondition')
+        elements.append('ModelCalibration')
+        elements.append('ModelInput')
+        elements.append('GeneralElements')
+        return elements
+
+    def has_all_required_elements(self):

```

Comment:
This function is not needed since you are not overriding the existing super class method.